src/main/java/net/minecraftforge/items/SlotItemHandler.java

```diff
@@ -126,7 +126,8 @@ public int getItemStackLimit(@Nonnull ItemStack stack)
     @Override
     public boolean canTakeStack(EntityPlayer playerIn)
     {
-        return !this.getItemHandler().extractItem(index, 1, true).isEmpty();
+        boolean vanillaResult = !this.getItemHandler().extractItem(index, 1, true).isEmpty();
+        return net.minecraftforge.event.ForgeEventFactory.onCanTakeItem(this.getStack(), this, this.inventory, playerIn, vanillaResult);
```

Comment:
The `inventory` field is meaningless in this class. You need to provide the `IItemHandler` to the event somehow. In fact perhaps the event should *only* accept `IItemHandler` and be wrapped for vanilla cases.